radix-common/src/data/manifest/model/manifest_resource_assertion.rs

```diff
@@ -326,17 +441,87 @@ impl GeneralResourceConstraint {
     }
 
     pub fn is_valid_for_fungible_use(&self) -> bool {
-        return self.required_ids.is_empty()
+        self.required_ids.is_empty()
             && self.lower_bound.is_valid_for_fungible_use()
             && self.upper_bound.is_valid_for_fungible_use()
             && self.allowed_ids.is_valid_for_fungible_use()
-            && self.is_valid_independent_of_resource_type();
+            && self.is_valid_independent_of_resource_type()
     }
 
     pub fn is_valid_for_non_fungible_use(&self) -> bool {
-        return self.lower_bound.is_valid_for_non_fungible_use()
+        self.lower_bound.is_valid_for_non_fungible_use()
             && self.upper_bound.is_valid_for_non_fungible_use()
-            && self.is_valid_independent_of_resource_type();
+            && self.is_valid_independent_of_resource_type()
+    }
+
+    pub fn validate_fungible(&self, amount: Decimal) -> Result<(), GeneralResourceConstraintError> {
+        self.validate_amount(amount)?;
+        // Static checker should have validated that there are no invalid non fungible checks
+        Ok(())
+    }
+
+    pub fn validate_non_fungible(
+        &self,
+        ids: &IndexSet<NonFungibleLocalId>,
+    ) -> Result<(), GeneralResourceConstraintError> {
+        self.validate_amount(Decimal::from(ids.len()))?;
+
+        for id in &self.required_ids {
+            if !ids.contains(id) {
+                return Err(GeneralResourceConstraintError::MissingRequiredNonFungible {
```

Comment:
OPTIONAL / MINOR: Slightly better would be to use `self.required_ids.is_subset(ids)` and return an error with all missing ids, using `self.required_ids.difference(ids).collect()` - and similar below, the other way around.